frontend/src/components/FilterPanel.tsx

```diff
@@ -118,25 +122,127 @@ export const FitlerPanel = () => {
     beds: 1,
     baths: 0.5,
   });
+  const [priceState, setPriceState] = useState({
+    min: 0,
+    max: 10000,
+  });
+  const [securityDepositState, setSecurityDepositState] = useState({
+    min: 0,
+    max: 10000,
+  });
+  const [applicationFeeState, setApplicationFeeState] = useState({
+    min: 0,
+    max: 10000,
+  });
+  const [sizeState, setSizeState] = useState({
+    min: 0,
+    max: 10000,
+  });
+  const [dateState, setDateState] = useState({
+    from: "",
+    to: "",
+  });
+
+  const applyFilters = () => {
+    const newFilters = {
+      availability: AvailabilityOptions[availabilityState],
+      housingAuthority: HousingAuthorityOptions[housingAuthorityState],
+      accessibility: JSON.stringify(
+        Array.from(accessibilityState).map((index) => AccessibilityOptions[index]),
+      ),
+      rentalCriteria: JSON.stringify(
+        Array.from(rentalCriteriaState).map((index) => RentalCriteriaOptions[index]),
+      ),
+      additionalRules: JSON.stringify(
+        Array.from(additionalRulesState).map((index) => AdditionalRulesOptions[index]),
+      ),
+      beds: bedBathState.beds.toString(),
+      baths: bedBathState.baths.toString(),
+      minPrice: priceState.min.toString(),
+      maxPrice: priceState.max.toString(),
+      minSecurityDeposit: securityDepositState.min.toString(),
+      maxSecurityDeposit: securityDepositState.max.toString(),
+      minApplicationFee: applicationFeeState.min.toString(),
+      maxApplicationFee: applicationFeeState.max.toString(),
+      minSize: sizeState.min.toString(),
+      maxSize: sizeState.max.toString(),
+      fromDate: dateState.from,
+      toDate: dateState.to,
+    };
+
+    setFilters({
+      ...newFilters,
+      approved: filters.approved,
+      sort: filters.sort ?? "",
+      search: filters.search ?? "0",
```

Comment:
```suggestion
      search: filters.search ?? "",
```

There seems to be a bug where clicking "Apply" sets the search value to "0"